File: 백준/Bronze/2798. 블랙잭/블랙잭.py
N, M = map(int, input().split())
lst = list(map(int, input().split()))

res = 0
for i in range(N-2):
    for j in range(i+1,N-1):
        for k in range(j+1,N):
            num = lst[i]+lst[j]+lst[k]
            if num > M:
                continue
            else:
                if res < num:
                    res = num
print(res)

File: 백준/Bronze/2798. 블랙잭/블랙잭.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const [N, M] = input.shift().split(" ");

let lst = input[0]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b);

let sum = 0;
let max = 0;

for (let i = 0; i < lst.length - 2; i++) {
  sum = 0;
  sum += lst[i];
  for (let j = i + 1; j < lst.length - 1; j++) {
    sum = lst[i];
    sum += lst[j];
    for (let k = j + 1; k < lst.length; k++) {
      if (max < sum + lst[k] && sum + lst[k] <= M) max = sum + lst[k];
    }
  }
}

console.log(max);
